### ISSUE TITLE
Auto-update pre-commit plugins

### ISSUE THREAD

**Issue by lukpueh** [secure-systems-lab/securesystemslib#734]
Plugins (most notably isort and black) in `.pre-commit-config.yaml` are not updated by Dependabot and thus prone to become out of sync with the same dev tools used by tox (also see discussion with @jku in #582).

We should find a way to keep them in-sync.

**Comment by jku** [comment]
without understanding pre-commit much, how about this:
* make `tox -e lint` fast (aka _use ruff_)
* start running `tox -e lint` in a commit hook

I think this would explicitly not use pre-commit itself though.

**Comment by lukpueh** [comment]
The difference would be that `tox -e lint` doesn't reformat your code.

**Comment by jku** [comment]
A `tox -e reformat` command would  be nice to have anyway maybe

**Comment by lukpueh** [comment]
I'm always pro lighter toolchain.

**Comment by L77H** [comment]
With Ruff as our linter now we can add `tox -e reformat` with `ruff format {[testenv:lint]lint_dirs}`. We could also add this immediately on `tox -e lint`, this would depend if the suggested reformatting changes should be reviewed first.

**Comment by lukpueh** [comment]
In my experience, reformatting and format checking + linting are usually separated. But I suppose we can combine them, now that both of them are so fast. The important thing is that `tox -e lint` fails CI, if the format is incorrect. What do others think?

**Comment by jku** [comment]
I think the autofix/autoformat feature is not a must-have but I'm not complaining if it's there. I wouldn't add it to `lint` itself though -- it feels like there should be a way to "just test", not fix.

In python-tuf I added `tox -e fix` which ends up running something like
```
ruff check --fix <DIRS>
ruff format <DIRS>
```

**Comment by lukpueh** [comment]
> I wouldn't add it to lint itself though -- it feels like there should be a way to "just test", not fix.

Agreed. Not so much because I want to be able just test, but more because I don't want change anyone's sources when they don't expect it.
